pkg/apiserver/apiserver.go

```diff
@@ -102,16 +111,57 @@ func (c completedConfig) New() (*NamespaceReservationServer, error) {
 		GenericAPIServer: genericServer,
 	}
 
+	accessor := meta.NewAccessor()
+	versionInterfaces := &meta.VersionInterfaces{
+		ObjectConvertor:  Scheme,
+		MetadataAccessor: accessor,
+	}
+	interfacesFor := func(version schema.GroupVersion) (*meta.VersionInterfaces, error) {
+		if version != admissionv1alpha1.SchemeGroupVersion {
+			return nil, fmt.Errorf("unexpected version %v", version)
+		}
+		return versionInterfaces, nil
+	}
+	restMapper := meta.NewDefaultRESTMapper([]schema.GroupVersion{admissionv1alpha1.SchemeGroupVersion}, interfacesFor)
+	restMapper.AddSpecific(
+		admissionv1alpha1.SchemeGroupVersion.WithKind("AdmissionReview"),
+		AdmissionsSchemeGroupVersion.WithResource("namespacereservations"),
+		AdmissionsSchemeGroupVersion.WithResource("namespacereservation"),
+		meta.RESTScopeRoot)
+
+	admissionReview := admissionreview.NewREST()
 	// TODO we're going to need a later k8s.io/apiserver so that we can get discovery to list a different group version for
 	// our endpoint which we'll use to back some custom storage which will consume the AdmissionReview type and give back the correct response
-	//apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(wardle.GroupName, registry, Scheme, metav1.ParameterCodec, Codecs)
-	//apiGroupInfo.GroupMeta.GroupVersion = v1alpha1.SchemeGroupVersion
-	//v1alpha1storage := map[string]rest.Storage{}
-	//apiGroupInfo.VersionedResourcesStorageMap["v1alpha1"] = v1alpha1storage
-	//
-	//if err := s.GenericAPIServer.InstallAPIGroup(&apiGroupInfo); err != nil {
-	//	return nil, err
-	//}
+	apiGroupInfo := genericapiserver.APIGroupInfo{
+		GroupMeta: apimachinery.GroupMeta{
+			GroupVersion:  AdmissionsSchemeGroupVersion,
+			GroupVersions: []schema.GroupVersion{AdmissionsSchemeGroupVersion},
+			SelfLinker:    runtime.SelfLinker(accessor),
+			RESTMapper:    restMapper,
+			InterfacesFor: interfacesFor,
+			InterfacesByVersion: map[schema.GroupVersion]*meta.VersionInterfaces{
+				admissionv1alpha1.SchemeGroupVersion: versionInterfaces,
+			},
+		},
+		VersionedResourcesStorageMap: map[string]map[string]rest.Storage{},
+		// TODO unhardcode this.  It was hardcoded before, but we need to re-evaluate
+		OptionsExternalVersion: &schema.GroupVersion{Version: "v1"},
+		Scheme:                 Scheme,
+		ParameterCodec:         metav1.ParameterCodec,
+		NegotiatedSerializer:   Codecs,
+		SubresourceGroupVersionKind: map[string]schema.GroupVersionKind{
```

Comment:
@smarterclayton this tricks the REST handler code into decoding the object we want, but it is a side-effect, not intent.  I think we should officially allow the intent.